sympy/parsing/latex/latex_parser.py

```diff
@@ -0,0 +1,128 @@
+import operator
+import os
+
+import sympy
+from sympy.external import import_module
+
+
+class LaTeXParsingError(Exception):
+    pass
+
+
+def _assert_nargs_func(args, n):
+    if len(args) != 3 + n:
+        raise LaTeXParsingError('latex function=%s expected %d args got %d' % (args[0].value, n, len(args)-3))
+
+
+def parse_latex(s):
+    r"""Converts the string ``s`` to a SymPy ``Expr``
+
+    Parameters
+    ==========
+
+    s : str
+        The LaTeX string to parse. In Python source containing LaTeX,
+        *raw strings* (denoted with ``r"``, like this one) are preferred,
+        as LaTeX makes liberal use of the ``\`` character, which would
+        trigger escaping in normal Python strings.
+
+    Examples
+    ========
+
+    >>> from sympy.parsing.latex import parse_latex
+    >>> expr = parse_latex(r"\frac {1 + \sqrt {\a}} {\b}")
+    >>> expr
+    (sqrt(a) + 1)/b
+    >>> expr.evalf(4, subs=dict(a=5, b=2))
+    1.618
+    """
+
+    # default behavior is to regenerate grammar from `LaTeX.lark` if
+    # `lark` is not importable it falls back to using the standalone
+    # precompiled lark latex grammar if it exists
+    _lark = import_module('lark')
+    if _lark is None:
+        # generated via lark standalone
+        # python -m lark.tools.standalone latex.lark > latex_grammar.py
+        standalone_lark = import_module('sympy.parsing.latex.latex_grammar')
+        if standalone_lark is None:
+            # TODO: throw proper error
+            raise ValueError('precompiled lark grammar "latex_grammar" not avialable')
+        parser = standalone_lark.Lark_StandAlone()
+        Transformer = standalone_lark.Transformer
+    else:
+        # TODO: should we use pkg_resource to get grammar file?  I
+        # think this would make sympy depend on setuptools which we
+        # would not like
+        with open(os.path.join(os.path.dirname(__file__), 'latex.lark')) as f:
+            latex_grammar = f.read()
+
+        parser = _lark.Lark(latex_grammar, parser='lalr',
+                            lexer='standard',
+                            propagate_positions=False,
+                            maybe_placeholders=False)
+        Transformer = _lark.Transformer
```

Comment:
@asmeurer if lark is available, it just looks like this!

Reply:
Wait so it's faster to use lark than the auto generated file if it is installed?

Reply:
No, I think the up-front cost of the standalone file will be faster. Successive calls would be identical, presumably. I have no benchmarks to support this.